src/main/java/com/synopsys/arc/jenkinsci/plugins/jobrestrictions/restrictions/job/RegexNameRestriction.java

```diff
@@ -86,7 +88,9 @@ public String getDisplayName() {
             return Messages.restrictions_Job_RegexName();
         }
 
+        @RequirePOST
         public FormValidation doCheckRegexExpression(@QueryParameter String regexExpression) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
```

Comment:
That permission check is not really needed. This method checks if the regex is valid. So it is not modifying anything or calls external systems. It is safe to just suppress the security warning